src/tools/server.ml

```diff
@@ -209,6 +209,7 @@ let make_server_thread
         Lwt.catch
           (fun () ->
            Lwt_unix.accept listening_socket >>= fun (plain_fd, cl_socket_address) ->
+           Lwt_unix.setsockopt listening_socket Unix.SO_KEEPALIVE true;
```

Comment:
we just get the (1980s) system defaults here. 
we may want to override the following as well
 *  TCP_KEEPCNT: overrides tcp_keepalive_probes
 *  TCP_KEEPIDLE: overrides tcp_keepalive_time
 *  TCP_KEEPINTVL: overrides tcp_keepalive_intvl

and make the whole thing configurable?

www.tldp.org/HOWTO/html_single/TCP-Keepalive-HOWTO

for future reference.